### ISSUE TITLE
rsyslog to h2 database connector

### ISSUE THREAD

**Issue by harringtonp** [rsyslog/rsyslog#1801]
In rsyslog I've created a template for a database insert statement called NV and I write these insert statements to a file ys_nv_haproxy.log as follows

`if $programname startswith 'haproxy' and $msg contains '/nextVehicles'    then action(type="omfile" file="/var/log/ys_nv_haproxy.log" template="NV")`

Then when the log is being rotated on a daily basis I set up postrotate to load the insert statements from ys_nv_haproxy.log into the h2 database as follows:

`java -cp "$dir/h2-1.4.196.jar:$H2DRIVERS:$CLASSPATH" org.h2.tools.RunScript -url jdbc:h2:tcp://127.0.0.1:7002/rsyslogdb -user rsysloguser -password rsyslogpwd -script /var/log/ys_nv_haproxy.log.1`

where I can view them using the h2 browser. Obviously with this it is a bulk load and I have to wait until the next day to view the previous days data.

Questions is is there an easy way of using the template NV to log the data immediately in the h2 database as soon as it appears in rsyslog ? I have seen rsyslog output modules for mysql and postgres and am looking for something similar that will work with my h2 setup. 

Basically if possible I'd like to be able to replace the

`action(type="omfile" file="/var/log/ys_nv_haproxy.log" template="NV")`

above with something while will connect to the h2 database and run the current insert statement

**Comment by davidelang** [comment]
I haven't heard of the h2 database before, pointers to it? (I'm not even going 
to try googling for "h2", that's just too short a string)

the odds are fairly low of this being a high priority (IIRC the postgres module 
was contributed, and it's far more common), so this would probably need to be 
contributed by someone.
k

**Comment by harringtonp** [comment]
h2 is an embedded java database like jetty. It is supposedly faster than Jetty but less feature rich. 

Best page for finding out about it is:

http://www.h2database.com/html/tutorial.html

I'm not necessarily looking for a driver, some way to take the template output as a script and pipe it into org.h2.tools.RunScript would also be good. I was hoping the h2 developers would have responded to

https://stackoverflow.com/questions/46444805/rsyslog-to-h2-database-connector

as a solution comes form their end as much as rsyslog.

**Comment by rgerhards** [comment]
I suggest you consider writing an external plugin in a scripting language you know: http://www.rsyslog.com/writing-external-output-plugins-for-rsyslog/
